cpp/src/arrow/compute/exec/plan_test.cc

```diff
@@ -1298,17 +1305,53 @@ TEST(ExecPlanExecution, ScalarSourceScalarAggSink) {
                 })
                 .AddToPlan(plan.get()));
 
-  ASSERT_THAT(
-      StartAndCollect(plan.get(), sink_gen),
-      Finishes(ResultWith(UnorderedElementsAreArray({
-          ExecBatchFromJSON(
-              {boolean(), boolean(), int64(), float64(), int64(), float64(), int64(),
-               float64(), float64()},
-              {ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR,
-               ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::ARRAY,
-               ArgShape::SCALAR},
-              R"([[false, true, 6, 5.5, 26250, 0.7637626158259734, 33, 5.0, 0.5833333333333334]])"),
-      }))));
+  auto exec_batch = ExecBatchFromJSON(
+      {boolean(), boolean(), int64(), int64(), float64(), int64(), float64(), int64(),
+       float64(), float64()},
+      {ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR,
+       ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR, ArgShape::SCALAR,
+       ArgShape::ARRAY, ArgShape::SCALAR},
+      R"([[false, true, 6, 6, 5.5, 26250, 0.7637626158259734, 33, 5.0, 0.5833333333333334]])");
+
+  ASSERT_THAT(StartAndCollect(plan.get(), sink_gen),
+              Finishes(ResultWith(UnorderedElementsAreArray({
+                  std::move(exec_batch),
+              }))));
+}
+
+TEST(ExecPlanExecution, ScalarSourceStandaloneNullaryScalarAggSink) {
+  ASSERT_OK_AND_ASSIGN(auto plan, ExecPlan::Make());
+  AsyncGenerator<std::optional<ExecBatch>> sink_gen;
+
+  BatchesWithSchema scalar_data;
+  scalar_data.batches = {
+      ExecBatchFromJSON({int32(), boolean()}, {ArgShape::SCALAR, ArgShape::SCALAR},
+                        "[[5, null], [5, false], [5, false]]"),
+      ExecBatchFromJSON({int32(), boolean()}, "[[5, true], [null, false], [7, true]]")};
+  scalar_data.schema = schema({
+      field("a", int32()),
+      field("b", boolean()),
+  });
+
+  auto sequence = Declaration::Sequence({
+      {"source", SourceNodeOptions{scalar_data.schema, scalar_data.gen(/*parallel=*/false,
+                                                                       /*slow=*/false)}},
+      {"aggregate", AggregateNodeOptions{/*aggregates=*/{
+                        {"count_all", "count(*)"},
+                    }}},
+      {"sink", SinkNodeOptions{&sink_gen}},
+  });
+
+  // index can't be tested as it's order-dependent
+  // mode/quantile can't be tested as they're technically vector kernels
+  ASSERT_OK(sequence.AddToPlan(plan.get()));
+
+  auto exec_batch = ExecBatchFromJSON({int64()}, {ArgShape::SCALAR}, R"([[6]])");
+
+  ASSERT_THAT(StartAndCollect(plan.get(), sink_gen),
+              Finishes(ResultWith(UnorderedElementsAreArray({
+                  std::move(exec_batch),
+              }))));
 }
```

Comment:
```suggestion
TEST(ExecPlanExecution, ScalarSourceStandaloneNullaryScalarAggSink) {
  BatchesWithSchema scalar_data;
  scalar_data.batches = {
      ExecBatchFromJSON({int32(), boolean()}, {ArgShape::SCALAR, ArgShape::SCALAR},
                        "[[5, null], [5, false], [5, false]]"),
      ExecBatchFromJSON({int32(), boolean()}, "[[5, true], [null, false], [7, true]]")};
  scalar_data.schema = schema({
      field("a", int32()),
      field("b", boolean()),
  });

  Declaration plan = Declaration::Sequence(
      {{"source",
        SourceNodeOptions{scalar_data.schema, scalar_data.gen(/*parallel=*/false,
                                                              /*slow=*/false)}},
       {"aggregate", AggregateNodeOptions{/*aggregates=*/{
                         {"count_all", "count(*)"},
                     }}}});
  ASSERT_OK_AND_ASSIGN(BatchesWithCommonSchema actual_batches,
                       DeclarationToExecBatches(std::move(plan)));

  auto expected = ExecBatchFromJSON({int64()}, {ArgShape::SCALAR}, R"([[6]])");
  AssertExecBatchesEqualIgnoringOrder(actual_batches.schema, actual_batches.batches,
                                      {expected});
}
```

We are slowly trying to move away from `StartAndCollect` to `DeclarationToXyz` methods in tests.  The latter is a bit more compact and reflects the public API (`StartAndCollect` is an internal method)  You don't need to fix old tests but new tests should be written with the new style.